pkg/api/types.go

```diff
@@ -747,6 +747,9 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
+	// Optional: The IP protocol for these endpoints. Supports "TCP" and
+	// "UDP".  Defaults to "TCP".
+	Protocol  Protocol `json:"protocol,omitempty"`
```

Comment:
Protocol needs to be pushed into the Endpoint struct that you created in #4580. I'm not sure this PR makes sense on its own.